__tests__/app.test.js

```diff
@@ -356,3 +356,98 @@ describe('api/articles/:article_id/comments', () => {
     });
   });
 });
+
+describe('api/articles (queries)', () => {
+  describe('GET - Successful Responses', () => {
+    test('status:200 and default sorts the articles by date descending', () => {
+      return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy('created_at', { descending: true });
+        });
+    });
+    test('status:200 and sorts the articles by author when specified', () => {
+      return request(app)
+        .get('/api/articles?sort_by=author')
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy('author', { descending: true });
+        });
+    });
+    test('status:200 and sorts the articles in ascending order when specified', () => {
+      return request(app)
+        .get('/api/articles?sort_by=author&order=ASC')
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy('author');
+        });
+    });
+    test('status:200 and filters the articles by specified topic', () => {
+      return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          console.log(articles);
+          expect(articles).toHaveLength(1);
+          articles.forEach((article) =>
+            expect(article).toEqual(
+              expect.objectContaining({
+                topic: 'cats',
+              })
+            )
+          );
+        });
+    });
+  });
+  describe('GET - Error Responses', () => {
+    test('status:400 if invalid sort by request (Bad request)', () => {
+      return request(app)
+        .get('/api/articles?sort_by=banana')
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Bad sort by request!');
+        });
+    });
+    test('status:400 if sort by query is spelt incorrectly (Bad request)', () => {
+      return request(app)
+        .get('/api/articles?sortby=author')
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe('Misspelt query!');
```

Comment:
probably wouldn't give a message as specific as that back - maybe it's just an invalid query

Reply:
I say that only because people could send a completely wrong query like `?grab=` and it isn't a spelling error just not a correct query

Reply:
Replaced this with invalid query